gpu/gpu.go

```diff
@@ -342,8 +349,31 @@ func GetGPUInfo() GpuInfoList {
 		}
 
 		rocmGPUs = AMDGetGPUInfo()
+
+		// Then Ascend
+		aHandles = initAscendHandles()
+
+		for i := range aHandles.deviceCount {
+			if aHandles.ascend != nil {
+				gpuInfo := AscendGPUInfo{
+					GpuInfo: GpuInfo{
+						Library: "ascend",
+					},
+					index: i,
+				}
+				C.ascend_bootstrap(*aHandles.ascend, C.int(i), &memInfo)
+				gpuInfo.TotalMemory = uint64(memInfo.total)
+				gpuInfo.FreeMemory = uint64(memInfo.free)
+				gpuInfo.ID = C.GoString(&memInfo.gpu_id[0])
+				gpuInfo.Name = C.GoString(&memInfo.gpu_name[0])
+				ascendGPUs = append(ascendGPUs, gpuInfo)
+
+				slog.Info(fmt.Sprintf("[%d] Name:%s: FreeMemory:%d, TotalMemory: %d", gpuInfo.ID, gpuInfo.Name, gpuInfo.FreeMemory, gpuInfo.TotalMemory))
```

Comment:
![image](https://github.com/user-attachments/assets/b6e35350-76e4-45d2-ad0a-c67456855c46)

Reply:
I need to set ASCEND_RT_VISIBLE_DEVICES, if this env is unset, ollama can't find any NPUs.